### PR TITLE
Make verifications suspendable.

### LINKED ISSUE #189: Make verifications suspendable
Currently, top-level object `SdJwtVerifier` exposes various verification methods as plain non-suspendable functions.

This is fine as long as the caller has obtained beforehand (or out-of-band) the public key of the issuer.
There are cases though, like those described in SD-JWT-VC profile (see [here](https://www.ietf.org/archive/id/draft-ietf-oauth-sd-jwt-vc-03.html#name-issuer-signed-jwt-verificat)), where issuer's public key should be 
- Fetched (this is the case for an issuer that exposes an SD-JWT-VC [metadata](https://www.ietf.org/archive/id/draft-ietf-oauth-sd-jwt-vc-03.html#name-jwt-vc-issuer-metadata) endpoint) or/and
- Resolved (this is the case of an issuer that uses DIDs).
 
For this reason and in order to support in the future SD-JWT-VC   the methods of `SdJwtVerifier` must be suspendable.
This will also affect also the interfaces of `JwtSignatureVerifier` and `KeyBindingVerifier`

### PR DESCRIPTION
**Pull request by dzarras**
Closes #189

### CODE REVIEW

File: src/main/kotlin/eu/europa/ec/eudi/sdjwt/SdJwtVerifier.kt [hunk 1]
```diff
@@ -119,7 +119,7 @@ fun interface JwtSignatureVerifier {
      * @return a new [JwtSignatureVerifier] that in addition applies to the
      *  extracted payload the [additionalCondition]
      */
-    fun and(additionalCondition: (Claims) -> Boolean): JwtSignatureVerifier = JwtSignatureVerifier { jwt ->
+    fun and(additionalCondition: suspend (Claims) -> Boolean): JwtSignatureVerifier = JwtSignatureVerifier { jwt ->
```
Comment (babisRoutis) [inline src/main/kotlin/eu/europa/ec/eudi/sdjwt/SdJwtVerifier.kt:122]: Do we really need a suspended lamda here?

File: src/test/kotlin/eu/europa/ec/eudi/sdjwt/Sample.kt [hunk 2]
```diff
@@ -66,7 +67,7 @@ fun genRSAKeyPair(): RSAKey =
         .issueTime(Date()) // issued-at timestamp (optional)
         .generate()
 
-fun main() {
+fun main() = runBlocking {
```
Comment (babisRoutis) [inline src/test/kotlin/eu/europa/ec/eudi/sdjwt/Sample.kt:70]: You can just use `suspend fun main` instead

**Comment by babisRoutis** [comment]
@dzarras  Perhaps we should rename milestone & target version to 0.4.0, since we are introducing a breaking change into the API

**Comment by dzarras** [comment]
> @dzarras Perhaps we should rename milestone & target version to 0.4.0, since we are introducing a breaking change into the API

Done. Milestone renamed and version bumped.

File: src/main/kotlin/eu/europa/ec/eudi/sdjwt/SdJwtVerifier.kt (hunk 1 above)
Comment (dzarras) [reply src/main/kotlin/eu/europa/ec/eudi/sdjwt/SdJwtVerifier.kt:122]: Not necessarily, but I added it for API flexibility.

**Review by babisRoutis — APPROVED** [review]

**Pull request merged**